pkg/server/http.go

```diff
@@ -252,6 +260,27 @@ func AdminAuth(expectedToken string) func(http.Handler) http.Handler {
 	}
 }
 
+// ARCAuth verifies that the ARC key is set in configuration before allowing requests.
```

Comment:
Hmmm.... I don't understand this logic. 

From what I understand (from ts version) the `/arc-ingest` is an ednpoint to handle ARC callbacks. 
So why we're checking only if the config exists, but we don't have any authentication for this callback?

Secondary, if we really want to have this endpoint accessible only if this token is set, then instead of this additional middleware, you could simply not register the endpoint in fiber if the value is empty (so in [line 195](https://github.com/bsv-blockchain/go-overlay-services/pull/128/files#diff-e05d42cd8f08393ade4bd51d3f8a8b0aab337392362fa317f450880d5f8f6725R195))